src/analyzer.ts

```diff
@@ -939,7 +944,8 @@ export class Analyzer {
    */
   private _checkForCyclesAndSortDependencies(
     config: LocallyValidScriptConfig | ScriptConfig | InvalidScriptConfig,
-    trail: Set<ScriptReferenceString>
+    trail: Set<ScriptReferenceString>,
```

Comment:
Nit: Might be a nicer callsite if you gave `trail` and `isDirectlyInvoked` default values, like `trail: Set<ScriptReferenceString> = new Set()`. Especially `trail`, but it also looks like `isDirectlyInvoked` is only potentially false when it's called recursively.

Reply:
I actually prefer not having a default, because it forces you to think about what the value should be at every call site. It might be easy to just forget to include it at all, otherwise.